### ISSUE TITLE
Implement a CalDAV sync adapter

### ISSUE THREAD

**Issue by tibbi** [SimpleMobileTools/Simple-Calendar#259]
The purpose would be making CalDAV synchronization work even without having Google Calendar enabled, which is handling the syncing for most people.

**Comment by logos88** [comment]
And what about making it compatible with DAVdroid? wouldn't be easier?

**Comment by tibbi** [comment]
it should already be compatible, the goal is not to have to install any third party app

**Comment by mdkuser** [comment]
It works with acaldav and davdroid, no need for google calendar!
I have disabled google calendar and it's working fine with acaldav.

BTW acaldav and davdroid are open source, so no need to reinvent the wheel. ;-)

**Comment by logos88** [comment]
I have tried Simple-Calendar with aCalDav in Lineageos (whitout any from google installed) and everything works great.

**Comment by MichaIng** [comment]
For me, repetitive events appear doubled, if I adjust a single one on any other device. Between caldav server and e.g. thunderbird, as well as android stock calendar, this works correctly.

**Comment by tibbi** [comment]
create a separate issue for it with some details please, to not spam this thread

**Comment by tibbi** [comment]
well the vast majority of people (which is my target audience) use their Google account for syncing and have Google Calendar installed, so it works fine to them. I cannot really detect if there is any kind of adapter installed, or what account they might want to use.

**Comment by SamWhited** [comment]
I appear to have Google Calendar installed and am logged into my Google account, so maybe it's a bug afterall then?

**Comment by tibbi** [comment]
not sure. If you open the Google Calendar, do you see any accounts/events in it?

**Comment by SamWhited** [comment]
I think I see the problem; Google Calendar is "Disabled", so presumably that means it's not registering its sync adapter. Either way, if you have no sync adapter installed this control saying "Syncing" and doing nothing seems very confusing.

**Comment by tibbi** [comment]
that can be, Ill take a look at it someday, if I can detect if there is an adapter available.

**Comment by cryptomilk** [comment]
I wouldn't start to support syncing. There are really good open source app out there doing the job very well. Focus just on the calendar component. I think that playground is already big enough :-)

**Comment by AgostinoSturaro** [comment]
I only had the Sony Xperia calendar app installed, and it turns out it can't properly synchronize shared calendars.
See here https://talk.sonymobile.com/t5/Xperia-E1/Shared-calendar-problem/td-p/873947

I thought it was your app misbehaving when I noticed this:
- I could add an event to the shared calendar from my phone using Simple Calendar, and then see it on my PC using Google Calendar
- vice-versa did not work, I could not create an event from my PC and see it in Simple Calendar

So I adopted the solution proposed by the user of the topic mentioned above,
I installed Google Calendar next to the Xperia Calendar, used Google Calendar to sync the shared calendar, and now I can see the shared events from all apps, including Simple Calendar.

>So I've installed google calendar, there was a passobility to check and activate an shared calendar,
so I did it.
>Now I can use shared calendar in preinstalled original calendar by Sony, and tin the google calnedar too.

Not the best solution, though. It would have been much easier if your app did the syncing as well. It seems it already does somewhat sync events, only one-way (phone to cloud).

I don't know if Samsung Phones have their own calendar app, nor if it works for shared calendars, but I imagine there can be similar problems there.

**Comment by jotaf98** [comment]
Hi, first of all superb job with the app! I'm a refugee from the TimeBlocks calendar app (since I found out they vacuum up all your calendar data and sell it to advertisers).

But I have to ask; if the Google Calendar app is a dependency (to enable syncing), why would a user prefer Simple Calendar at all?

Google Calendar is still uploading your non-Google calendars to the mothership in the background, so what is the added value of Simple Calendar in that case? Feature-wise, they are similar; for me the added value is the privacy and confidence that it's open-source. If I have to re-enable Google Calendar, that value is lost.

I did try the WebDav adapters (DAVdroid and aCalDav). I had to jump through hoops (enable unsigned apps, install F-Droid, install and configure those), and couldn't get either to work with Google Calendar (they're not very friendly -- see solutions [here](https://android.stackexchange.com/questions/191252/how-to-synchronize-gmail-calendars-in-davdroid-1-10-1-gplay-using-caldav-in-and), yes they're complicated, no they didn't work for me).

In conclusion, I don't expect an average user to do any of that even if it worked, and if the average user is just going to keep the Google Calendar app installed, then opening Simple Calendar over Google Calendar is just a cosmetic choice (privacy is still undermined).

With that said, I understand that it's a tough feature to implement and I wish other people did not rely on Google Calendars so much for shared events, but unfortunately it's the de facto standard in many places.

**Comment by tibbi** [comment]
This issue is about non-google services. I dont think it is possible to sync with Google servers with my own solution at all. It it meant more for people using for example NextCloud as the server (instead of Google) and using DAVDroid for syncing between the server and the device. Anyway I wont be working on a custom sync adapter anytime soon, I pasted it here just as an idea for the future.

**Comment by jotaf98** [comment]
Thanks. My frustration should be directed at Google for making interoperability with open standards (WebDav) so unreliable; it seems to work for many other hosts. I hope that it gets better in the future, as it's so pervasive.

In case it's useful at some point, the [official calendar API](https://developers.google.com/calendar/) seems a better bet for this purpose than DAV -- my previous calendar app was using this API and was very easy to set up and reliable (I know it was using it because there is a list of apps with API permission in the Google Calendar settings webpage).

**Comment by sigvaldm** [comment]
Do I understand it correctly that it should be possible to use aCalDav to sync Simple Calendar Pro to a CalDAV calendar hosted at, say, NextCloud or on an own server? Without going through the built-in calendar or any Google products, thereby compromising privacy? If so I can't figure out how. I installed Simple Calendar (non-Pro to test) and aCalDav, but see no signs of adding a CalDAV adress anywhere. In fact, aCalDAV doesn't show up anywhere, not even on the phone's menu. I am strongly in favor of a calendar app which supports custom CalDAV servers, and am willing to pay for Simple Calendar Pro, if it supports this and seems to work smoothly.

**Comment by tibbi** [comment]
ye, since Simple Calendar is offline and will stay offline, you will need a third party "sync adapter" for taking care of the syncing between your device and the servers. APps like ACalDAV or DavDroid seem to work fine. Exact steps for setting them up depend on the exact service you use. You should find some acaldav related url in nextcloud, it might contain "webdav" or so.

**Comment by TjrGithub** [comment]
It works fine on LineageOS 15.1 with davx5 (fdroid version) and mailbox.org.

**Comment by roynico** [comment]
For me, it works well both with DAVx abd ACalDAV.

**Comment by tibbi** [comment]
so yeah, Ill close this. It is way too advanced and would require internet access.
